ferrocene/doc/sphinx-shared-resources/exts/ferrocene_domain_cli/domain.py

```diff
@@ -47,6 +47,32 @@ def run(self):
 
         return [node]
 
+class SubcommandDirective(ObjectDescription):
+    has_content = True
+    required_arguments = 1
+    final_argument_whitespace = True
+
+    def handle_signature(self, sig, signode):
+        signode += addnodes.desc_name("", sig)
+
+    def add_target_and_index(self, name_cls, sig, signode):
+        if PROGRAM_STORAGE not in self.env.temp_data:
+            warn("cli:subcommand outside cli:program isn't supported", self.get_location())
+            program_storage = ProgramStorage("PLACEHOLDER", False)
+        else:
+            program_storage: ProgramStorage = self.env.temp_data[PROGRAM_STORAGE]
+
+        subcommand = Subcommand(
+            self.env.docname,
+            program_storage,
```

Comment:
I am noting that here we pass `program_storage`, while for `Option` we pass `program_storage.name`. https://github.com/ferrocene/ferrocene/blob/a6cc042d73d53be95dab510276c6a0eb8f15c3d3/ferrocene/doc/sphinx-shared-resources/exts/ferrocene_domain_cli/domain.py#L92-L97

For both we can probably just pass the `program_storage` and extract the `name` and the `no_traceability_matrix` in the constructor.

Reply:
I actually renamed "program" to "program_name" for clarity.